packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1607,6 +1637,15 @@ abstract class Element implements BuildContext {
     });
   }
 
+  bool _debugIsInScope(Element target) {
+    assert(target != null);
+    if (target == this)
+      return true;
+    if (_parent != null)
+      return _parent._debugIsInScope(target);
+    return false;
+  }
```

Comment:
I probably would have written this iteratively rather than recursively but whatever.